tests/cupy_tests/core_tests/test_ndarray_reduction.py

```diff
@@ -216,13 +216,14 @@ def test_ptp_nan_imag(self, xp, dtype):
     'order': ('C', 'F'),
 }))
 @testing.gpu
-@unittest.skipIf(cupy.cuda.cub_enabled is False, 'The CUB module is not built')
-class TestCUBreduction(unittest.TestCase):
+@unittest.skipUnless(
+    cupy.core._backend.get_routine_backends() == ['cub'],
```

Comment:
No, another backend may be chosen. `cupy.core._backend.get_routine_backends()[:1] == ['cub']` may be better because the `get_routine_backends` indicates the priority of backends.

Reply:
Ah I see. That sounds good. Either `[0]` or `[:1]` is fine.

Reply:
Hi @asi1024 I thought about this again, and I think the rationale here was that if `cupy.cuda.cub` is built it should get tested (regardless how backends were set). If one of the CIs would be set up with `CUPY_BACKENDS="cub,"` then this is OK; otherwise, once #2584 is up, we'll need to ensure these tests are run (by removing `skipUnless()` and using  `setUp()` and `tearDown()` to temporarily overwrite the backends).

Reply:
I agree to set backends in `setUp` and `tearDown`. Fixed in 43b2da6.